pkg/utils/release/controller.go

```diff
@@ -109,29 +112,107 @@ func (s *SuiteController) GetPipelineRunInNamespace(namespace, releaseName, rele
-		Name:      name,
-		Namespace: namespace,
+// DeleteReleasePlan deletes the releasePlan resource with the given name from the given namespace.
+// Optionally, it can avoid returning an error if the resource did not exist:
+// specify 'false', if it's likely the ReleasePlan has already been deleted (for example, because the Namespace was deleted).
+func (s *SuiteController) DeleteReleasePlan(releasePlanName, namespace string, reportErrorOnNotFound bool) error {
+	releasePlan := appstudiov1alpha1.ReleasePlan{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      releasePlanName,
+			Namespace: namespace,
+		},
+	}
+	err := s.KubeRest().Delete(context.TODO(), &releasePlan)
+	if err != nil && !reportErrorOnNotFound && k8sErrors.IsNotFound(err) {
+		err = nil
 	}
+	return err
+}
+
+// CreateReleasePlanAdmission creates a new ReleasePlanAdmission using the given parameters.
+func (s *SuiteController) CreateReleasePlanAdmission(name, devNamespace, application, managedNamespace, AutoReleaseLabel, releaseStrategy string) (*appstudiov1alpha1.ReleasePlanAdmission, error) {
 
-	releaseLink := &v1alpha1.ReleaseLink{}
-	err := s.KubeRest().Get(context.TODO(), namespacedName, releaseLink)
-	if err != nil {
-		return nil, err
+	var releasePlanAdmission *appstudiov1alpha1.ReleasePlanAdmission = &appstudiov1alpha1.ReleasePlanAdmission{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      name,
+			Namespace: managedNamespace,
+		},
+		Spec: appstudiov1alpha1.ReleasePlanAdmissionSpec{
+			DisplayName: name,
+			Application: application,
+			Origin: kcp.NamespaceReference{
+				Namespace: devNamespace,
+			},
+			Environment:     "test-environment",
+			ReleaseStrategy: releaseStrategy,
+		},
+	}
+
+	if AutoReleaseLabel != "" { // AutoReleaseLabel in ReleasePlan is not missing
+		releasePlanAdmission.ObjectMeta.Labels[appstudiov1alpha1.AutoReleaseLabel] = AutoReleaseLabel
+	}
+	return releasePlanAdmission, s.KubeRest().Create(context.TODO(), releasePlanAdmission)
+}
+
+// GetReleasePlanAdmission returns the releasePlan with the given name in the given namespace.
+func (s *SuiteController) GetReleasePlanAdmission(releasePlanAdmissionName, managedNamespace string) (*appstudiov1alpha1.ReleasePlanAdmission, error) {
+	releasePlanAdmission := &appstudiov1alpha1.ReleasePlanAdmission{}
+
+	err := s.KubeRest().Get(context.TODO(), types.NamespacedName{
+		Name:      releasePlanAdmissionName,
+		Namespace: managedNamespace,
+	}, releasePlanAdmission)
+
+	return releasePlanAdmission, err
+}
+
+// DeleteReleasePlanAdmission deletes the releasePlanAdmission resource with the given name from the given namespace.
+// Optionally, it can avoid returning an error if the resource did not exist:
+//  specify 'false', if it's likely the ReleasePlan has already been deleted (for example, because the Namespace was deleted)
+func (s *SuiteController) DeleteReleasePlanAdmission(releasePlanAdmissionName, managedNamespace string, reportErrorOnNotFound bool) error {
```

Comment:
I think this should just be called namespace

Reply:
Agreed, fixed